src/openrct2/core/IStream.cpp

```diff
@@ -12,9 +12,6 @@
 #include "../object/Object.h"
 #include "Memory.hpp"
 #include "String.hpp"
-
-#include <vector>
```

Comment:
This file does use vector (e.g. on line 22/19), so I don’t think this should be removed.

Reply:
Yeah but it's included in the associated header IStream.hpp

Reply:
Shouldn’t files carry their own dependencies, rather than relying on other files to do it for them?

Reply:
I would say so expect when there is a directly associated header. This is the implementation of the istream header and therefore i don't really see the point in having it in both it's just increasing compile times for no reason.

Reply:
The include in the .hpp file seems to be unused. It'd be better to move it to the source file.

Reply:
It's used on L236 and afaik you aren't meant to forward declare std classes

Reply:
Ah yup, you're right. I'd still argue that the #include in the source file is nice to have, since std::vector gets used as an implementation detail outside of function declarations and data members. I don't expect the extra inclusion of system headers to have a real impact on build time.

Reply:
Its best to always include the headers you are using, if you rely on the include in another header and that gets refactored you need to change unrelated things.